lib/app/features/config/providers/force_update_provider.c.dart

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ion/app/features/config/providers/config_provider.c.dart';
+import 'package:ion/app/features/config/providers/force_update_last_sync_date_provider.c.dart';
+import 'package:ion/app/features/config/providers/force_update_util_provider.c.dart';
+import 'package:ion/app/features/core/providers/app_lifecycle_provider.c.dart';
+import 'package:ion/app/features/core/providers/env_provider.c.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'force_update_provider.c.freezed.dart';
+part 'force_update_provider.c.g.dart';
+
+@freezed
+class ForceUpdateState with _$ForceUpdateState {
+  const factory ForceUpdateState({
+    @Default(false) bool shouldShowUpdateModal,
+  }) = _ForceUpdateState;
+
+  const ForceUpdateState._();
+}
+
+@Riverpod(keepAlive: true)
+class ForceUpdate extends _$ForceUpdate {
+  @override
+  ForceUpdateState build() {
+    _checkAndUpdateConfig();
```

Comment:
use `fireImmediately` for the `ref.listen` instead of calling this method manually.